sn_transfers/src/wallet/hot_wallet.rs

```diff
@@ -429,7 +443,7 @@ impl HotWallet {
             start.elapsed()
         );
         debug!("Available CashNotes: {:#?}", available_cash_notes);
-        let reason_hash = Default::default();
+        let reason_hash = Hash::hash(b"SPEND_REASON_FOR_STORAGE");
```

Comment:
Can this be renamed into `input_reason_hash` along with the arguments to `OfflineTransfer::new()` to better distinguish between the "input reason" and "output purpose".

Also if possible can we change `reason -> purpose` if they're for Outputs.